safehttp/mux.go

```diff
@@ -168,6 +182,25 @@ func (m methodHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	h.ServeHTTP(w, r)
+}
+
+// handlerWithInterceptors encapsulates a handler and its corresponding
+// interceptors.
+type handlerWithInterceptors struct {
+	handler        Handler
+	configs        map[string]Config
+	interceps      map[string]Interceptor
```

Comment:
(Why?) Is it required to keep a map here?

Couldn’t you turn the interceptors into an ordered slice (with their config) and avoid the map altogether in the `handlerWithInterceptors` type?

For serving, I don’t see the need, and a map lookup is more expensive than just iterating through a slice.

Reply:
Ah, I see that kele raced me to it with the appliedInterceptor remark.